tmt/steps/__init__.py

```diff
@@ -1958,7 +1958,16 @@ def _login(
             # Attempt to push the workdir to the guest
             try:
                 guest.push()
-                cwd = cwd or self.parent.plan.worktree
+                worktree = self.parent.plan.worktree
+                test_path = self.parent.plan.discover.tests(
+                    )[-1].path if self.parent.plan.discover.tests() else None
+
+                if worktree and test_path:
+                    test_path = worktree.parent / "discover" / test_path.unrooted()
+                else:
+                    test_path = worktree
+
+                cwd = cwd or test_path
```

Comment:
All these new lines above would be wasted if `cwd` is set. How about putting them under `if not cwd`?